### PR TITLE
Fix podman_tooling script

### PR DESCRIPTION
**Pull request by cevich**
***Depends on:*** https://github.com/containers/automation_images/pull/134

This script is needed by both podman and buildah CI.  However a bug in
the line `export GOPATH="${GOPATH:/var/tmp/go}"` was causing it to
silently exit before doing it's job.  Fortunately the downstream build
systems are somewhat tolerant, so the problem wasn't noticed at image
build time.

Simplify the script by requiring it to be called under $SUDO, this
allows moving a bunch of golang variable definitions into the script and
also making it `set -e` so failures are noticed sooner.

Ref: Silent failure example: https://cirrus-ci.com/task/6089758945312768?logs=main#L1905-L1906

### CODE REVIEW

**Review by lsm5 — APPROVED** [review]
LGTM

**Comment by cevich** [comment]
force-push: rebased on @lsm5 #134 to make sure my `podman_tooling.sh` changes work.

**Comment by lsm5** [comment]
`lilto` command not found

File: cache_images/fedora_packaging.sh [hunk 1]
```diff
@@ -198,10 +198,5 @@ fi
 # It was observed in F33, dnf install doesn't always get you the latest/greatest
 lilto $SUDO dnf update -y
 
-echo "Configuring Go environment"
-export GOPATH=/var/tmp/go
-mkdir -p "$GOPATH"
-eval $(go env | tee /dev/stderr)
-export PATH="$GOPATH/bin:$PATH"
 # shellcheck source=./podman_tooling.sh
```
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:201]: Probably don't need this `shellcheck` any more

File: cache_images/podman_tooling.sh [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
 
 echo "Installing runtime tooling"
-export GOPATH="${GOPATH:/var/tmp/go}"
-export GOSRC=/var/tmp/go/src/github.com/containers/podman
-export GOCACHE="${GOCACHE:-/root/.cache/go-build}"
 lilto git clone --quiet https://github.com/containers/podman.git "$GOSRC"
 
 cd "$GOSRC" || die "Podman repo. not cloned to expected directory: '$GOSRC'"
 # Calling script already loaded lib.sh
 # shellcheck disable=SC2154
```
Comment (edsantiago) [inline cache_images/podman_tooling.sh:29]: Is this `shellcheck` still needed? (I see the same 2154 number below, next to a variable, so I'm guessing 2154 has to do with variables)

File: cache_images/podman_tooling.sh [hunk 3]
```diff
@@ -24,15 +38,15 @@ else  # Fedora
...
-    $SUDO make install.modules-load
+    make install.modules-load
 fi
 
 # Make pristine for other runtime usage/expectations also save a bit
 # of space in the images.
-$SUDO rm -rf "$GOPATH/src" "$GOCACHE"
-$SUDO chown -R root.root /var/tmp/go
+rm -rf "$GOPATH/src" "$GOCACHE"
+chown -R root.root /var/tmp/go
```
[the remark this answers is not in the document]
Comment (cevich) [reply cache_images/podman_tooling.sh:52]: yep.

**Comment by cevich** [comment]
> All the sudo bash makes me cringe. Why not just add a hashbang-bash line to the script, make it 755, and invoke it directly? 

I was afraid of making it appear like users should ever try and execute that script.  I s'pose I could make it executable in the setup script.

> Also, CI is failing with a "kubik" error; sounds like the Ubuntu issue from a week or two ago.

This is expected since this PR depends on https://github.com/containers/automation_images/pull/134

File: cache_images/fedora_packaging.sh (hunk 1 above)
Comment (cevich) [reply cache_images/fedora_packaging.sh:201]: okay I'll try.

**Review by edsantiago — APPROVED** [review]

**Comment by edsantiago** [comment]
6b66a74 LGTM (I did not review 0220c22)

**Comment by cevich** [comment]
Status note: Waiting for https://github.com/containers/automation_images/pull/134 to merge, then this can go in.

**Pull request merged**